CRM/Price/BAO/LineItem.php

```diff
@@ -819,60 +825,107 @@ protected function _getFinancialItemsToRecord($entityID, $entityTable, $contribu
 
     // gathering necessary info to record negative (deselected) financial_item
     $updateFinancialItem = "
-  SELECT fi.*, SUM(fi.amount) as differenceAmt, price_field_value_id, financial_type_id, tax_amount
+  SELECT fi.*, price_field_value_id, financial_type_id, tax_amount
```

Comment:
There is no use of it as we grouping on basis of ```fi.id```

Reply:
AH ok - you avoid reverting multiple times by keying on line item with this $financialItemsArray[$updateFinancialItemInfoValues['entity_id']]

It's worth noting that on ?ubuntu? you can't rely on natural sort to put the highest id last (which I think you are relying on - but perhaps the group by will cause it to sort like that - I wonder how we can check - I know Jaap was using the system without the natural sort from some test bugs that he hit)

Reply:
@eileenmcnaughton the ```differenceAmt``` was an alias used in old code to calculate the sum of all financial amount that is related to a line-item, this difference amount is used to decide the surplus or due amount so to make an adjustment on changeFeeSelections. But later to handle adjustment at more detail esp. for mult-line-item contribution we started to fetch result on basis of financia_item.id. So then I realized that there's no use of the sum now as it represents each financial_item's amount. So yes it's a tidy up

Reply:
a tidy up - but in the end I concluded that it was necessary to stop using it to make the fix work